moped-editor/src/views/projects/projectView/ProjectMilestones.js

```diff
@@ -28,14 +27,167 @@ import {
 import { useMutation } from "@apollo/client";
 import { format } from "date-fns";
 import parseISO from "date-fns/parseISO";
-import Autocomplete from "@mui/material/Autocomplete";
 
-// Helpers
 import { phaseNameLookup } from "src/utils/timelineTableHelpers";
-import DateFieldEditComponent from "./DateFieldEditComponent";
 import ToggleEditComponent from "./ToggleEditComponent";
-import ButtonDropdownMenu from "../../../components/ButtonDropdownMenu";
-import MilestoneTemplateModal from "./MilestoneTemplateModal";
+import MilestoneTemplateModal from "./ProjectMilestones/MilestoneTemplateModal";
+import MilestoneAutocompleteComponent from "./ProjectMilestones/MilestoneAutocompleteComponent";
+import DataGridDateFieldEdit from "./ProjectMilestones/DataGridDateFieldEdit";
+import DeleteConfirmationModal from "./DeleteConfirmationModal";
+
+const useMilestoneNameLookup = (data) =>
+  useMemo(() => {
+    if (!data) {
+      return {};
+    }
+    return data.moped_milestones.reduce(
+      (obj, item) =>
+        Object.assign(obj, {
+          [item.milestone_id]: item.milestone_name,
+        }),
+      {}
+    );
+  }, [data]);
+
+const useColumns = ({
+  data,
+  rowModesModel,
+  handleEditClick,
+  handleSaveClick,
+  handleCancelClick,
+  handleDeleteOpen,
+  milestoneNameLookup,
+}) =>
+  useMemo(() => {
+    return [
+      {
+        headerName: "Milestone",
+        field: "milestone_id",
+        renderCell: ({ row }) => row.moped_milestone?.milestone_name,
+        // input validation:
+        preProcessEditCellProps: (params) => ({
+          ...params.props,
+          error: !params.props.value,
+        }),
+        editable: true,
+        renderEditCell: (props) => (
+          <MilestoneAutocompleteComponent
+            {...props}
+            milestoneNameLookup={milestoneNameLookup}
+          />
+        ),
+        width: 250,
+      },
+      {
+        headerName: "Description",
+        field: "description",
+        width: 200,
+        editable: true,
+        renderEditCell: (props) => <DataGridTextField {...props} />,
+      },
+      {
+        headerName: "Related phase",
+        field: "moped_milestone",
+        editable: false, // would it be cool for this to update when someone edits the milestone
+        valueGetter: (value) => {
+          return phaseNameLookup(data)[value?.related_phase_id] ?? "";
```

Comment:
i agree! ill work on it

Reply:
thank you so much for bringing this up, because we already had a hook and now using this I dont have to pass data as a dependency to the useColumns function